flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimLoader.java

```diff
@@ -42,8 +43,10 @@
 	private HiveShimLoader() {
 	}
 
-	public static HiveShim loadHiveShim() {
-		String version = HiveVersionInfo.getVersion();
+	public static HiveShim loadHiveShim(String version) {
+		if (StringUtils.isNullOrWhitespaceOnly(version)) {
```

Comment:
I wonder if we can move the default version assignment up the stack and as early as possible? HiveShim should be just dealing with whatever version it's assigned to rather than deciding the default version for callers.